torch/quantization/quantize.py

```diff
@@ -107,8 +114,13 @@ def add_observer_(module, non_leaf_module_list=None, device=None):
             if hasattr(child, 'qconfig') and child.qconfig is not None:
                 child.add_module('activation_post_process', child.qconfig.activation())
                 child.register_forward_hook(_observer_forward_hook)
+
+                # Attaching prehook
+                if prehook is not None:
+                    child.add_module('activation_pre_process', prehook())
+                    child.register_forward_pre_hook(_observer_forward_pre_hook)
         else:
-            add_observer_(child, non_leaf_module_list, device)
+            add_observer_(child, non_leaf_module_list, device, prehook)
```

Comment:
This should not be needed. observer itself is a hook, so not sure why we need another hook.

Reply:
this is because the current implementation here is using the expected input for bias correction (stuff from appendix B), the second hook is to grab the inputs where as the first hook is to get the output

trying to refactor the bias correction logic to not need the appendix B stuff (getting rid of that will get rid of the need for the prehook)